### ISSUE TITLE
add a level creator maybe?

### ISSUE THREAD

**Issue by sansundertalereal** [ipodtouch0218/NSMB-MarioVsLuigi#94]
the same 8 levels do get a bit tiring after a bit
a way to spice everything up a bit would to be to add a level creator, being able to add enemies, blocks, and all of that
I would think you would also be able to decide the level height and length, and where the stars spawn
and maybe those special features, like the platforms in the Bricks level?
also idk how to tag this as new feature so uhhhhhh...

**Comment by SLG64** [comment]
This would be a great feature, especially because people could see others creativity and have a chance to explore their verson of Mario.

**Comment by DillyzThe1** [comment]
if you wanna know how to add custom levels, then allow me to show you!

## How to make a custom level!
### Setup & Game Recognition
first off, get the game compiling. you can do this by reading the game's [unity editor tutorial here](https://github.com/ipodtouch0218/NSMB-MarioVsLuigi#opening-in-unity-editor).
next, you have to enter the MainMenu scene and clicking on MainMenuManager.
![image](https://user-images.githubusercontent.com/34419642/189546010-19364b97-e634-47d2-af72-e038be5a2b4b.png)
now, find <b>Level Camera Positions</b> and add a new element.
![image](https://user-images.githubusercontent.com/34419642/189546031-5059618d-426a-4683-9aad-d49e7f2937f8.png)
then, you have to drag the camera position you want. you can leave this default, but i recommend doing grassland's camera position if you don't have one.
![image](https://user-images.githubusercontent.com/34419642/189546051-297a2b80-ba8d-40b2-99dd-3305b9989402.png)
next, find the maps array and add a new element containing the name of your level. i will be doing `[M] Plains` because i want M to stand for modded & i'm using plains as the name for the map.
![image](https://user-images.githubusercontent.com/34419642/189546106-b4ae7cd1-e8d2-4855-bc39-ab2355b5b99b.png)
At this point, you want to open the Template folder and copy `LevelTemp` under a new name (your level name)
![image](https://user-images.githubusercontent.com/34419642/189546154-172c2a70-1d1e-4744-b0ee-65e38047ad96.png)
put this in the levels folder, then double click the file
finally, go to _File > Build Settings_ and drag your level file under the last level name there
![image](https://user-images.githubusercontent.com/34419642/189546260-37b6e6ad-4aa2-4670-9d55-79be9e7e31d2.png)
hooray! you can now use _File > Build And Run_ to export the game and play the template level under your new name with friends in private lobbies!
### Editing A Level
go to _Window > 2D > Tile Palette_
next, in your scene file hit the dropdown on Grid & select Tilemap_Ground
then, choose a palette in the Tile Palette window and click any tile! 
_Note: you can also drag your mouse over multiple to paint stuff like pipes!_
finally, start painting tiles all over the place!
you can now hit control + s to save and hit _File > Build And Run_ to play!

**Comment by SLG64** [comment]
What if you're on a Chromebook that cannot access Linux?

**Comment by DillyzThe1** [comment]
this tutorial is obviously meant for people actually able to compile the game, so sorry if you're on web builds by force :/
also note that i did this on windows and you jsut need a standard desktop pc or laptop (windows, linux, mac)
(also, fun fact: ChromeOS is actually linux-based. you just can't really do almost anything a linux user can.)

**Comment by sansundertalereal** [comment]
can you play on the custom levels online?
I assume not, right?

**Comment by DillyzThe1** [comment]
you can play them online with friends who have the mod
and yes, i've been testing this stuff all day with friends

**Comment by SLG64** [comment]
Is there an alternative for the Unity editor?

**Comment by DillyzThe1** [comment]
no, the game was written inside of unity and heavily relies on the engine's abilities & functionality to work due to how this stuff functions anyway
i have been considering making a level editor in my mod, and may consider publishing the mod if demand is met

the mod also currently includes:
- the ability to disable certain powerups through a nice little menu
- a whole new map
- a bunch of inside jokes between my friends

i certaintly wouldn't mind rebuilding this mod with what i know for the public without the inside jokes (or just making them a toggle) as i do plan to add a level editor for my friends aswell

**Comment by SLG64** [comment]
Would I possibly be able to contribute to your mod?

**Comment by DillyzThe1** [comment]
the mod is a private build that simply exists between me and my friends for the sake of pure fun
if people really want, i'll do the public recreations of the useful features i have made and planned as a fork of this repo

**Comment by DillyzThe1** [comment]
oh, and if i do publicly release the mod, you could show off levels in the discussions tab the fork would have and i would consider downloading the json (which is what i'll save the level as when i make the editor) and putting it in the mod for people to play with

**Comment by SLG64** [comment]
Ok, I guess I wouldn't be much help seeing as I don't know much about coding or anything close to it as I only have a Chromebook at the moment.

**Comment by SLG64** [comment]
My discord isn't working at the moment, but I can give you my email that I use for my YouTube: matthewmonahan703@gmail.com

**Comment by DillyzThe1** [comment]
alright, i'll try to remember to send an email when the mod gets released. (but if you ever get your discord account working, PLEASE reach out to me about that. it's most efficient to contact me there.)
i'll create the forked repository by today at 5pm cst, so if you have ANY suggestions/concepts, please send them to that repository's issues or discussions

**Comment by SLG64** [comment]
okie dokie
I will
